### PR TITLE
Separate releases

### PR DESCRIPTION
**Pull request by FreePhoenix888**
- Add publish-release.sh
- Add csharp-publish-release.sh
- Add cpp-publish-release.sh

### CODE REVIEW

File: MultiProjectRepository/publish-release.sh [hunk 1]
```diff
@@ -1,59 +1,34 @@
...
 fi
 
-PACKAGE_RELEASE_NOTES_STRING=$(jq -saR . <<< "$PACKAGE_RELEASE_NOTES_STRING_BUFFER")
-echo $PACKAGE_RELEASE_NOTES_STRING
+RELEASE_BODY_STRING=$(jq -saR . <<< "$RELEASE_BODY")
+echo "RELEASE_BODY_STRING: $RELEASE_BODY_STRING"
 
 curl --request POST \
---url "https://api.github.com/repos/$GITHUB_REPOSITORY/releases" \
+--url "https://api.github.com/repos/$REPOSITORY_NAME/releases" \
```
Comment (Konard) [inline MultiProjectRepository/publish-release.sh:24]:
```suggestion
--url "https://api.github.com/repos/$GITHUB_REPOSITORY/releases" \
```

File: MultiProjectRepository/csharp-publish-release.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+then
+    echo "CSHARP_PACKAGE_VERSION.txt not found."
+    exit 1
+fi
+
+PACKAGE_VERSION=$(<CSHARP_PACKAGE_VERSION.txt)
+PACKAGE_RELEASE_NOTES=$(<CSHARP_PACKAGE_RELEASE_NOTES.txt)
+PACKAGE_URL="https://www.nuget.org/packages/Platform.$REPOSITORY_NAME/$CSHARP_PACKAGE_VERSION"
+
+TAG="$PACKAGE_VERSION"
```
Comment (Konard) [inline MultiProjectRepository/csharp-publish-release.sh:18]:
```suggestion
TAG="csharp_$PACKAGE_VERSION"
```

File: MultiProjectRepository/cpp-publish-release.sh [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+then
+    echo "CPP_PACKAGE_VERSION.txt not found."
+    exit 1
+fi
+
+PACKAGE_VERSION=$(<CPP_PACKAGE_VERSION.txt)
+PACKAGE_RELEASE_NOTES=$(<CPP_PACKAGE_RELEASE_NOTES.txt)
+PACKAGE_URL="https://www.nuget.org/packages/Platform.$REPOSITORY_NAME.TemplateLibrary/$PACKAGE_VERSION"
+
+TAG="$PACKAGE_VERSION"
```
Comment (Konard) [inline MultiProjectRepository/cpp-publish-release.sh:18]:
```suggestion
TAG="cpp_$PACKAGE_VERSION"
```

File: MultiProjectRepository/publish-release.sh [hunk 4]
```diff
@@ -1,59 +1,34 @@
...
+echo "Repository name: $REPOSITORY_NAME";
+echo "Tag: $TAG";
+echo "Default branch: $DEFAULT_BRANCH";
+echo "Release name: $RELEASE_NAME";
+echo "Release body: $RELEASE_BODY";
 
-TAG_BUFFER=""
-NAME_BUFFER=""
-PACKAGE_RELEASE_NOTES_STRING_BUFFER=""
+tag_id=$(curl --request GET --url "https://api.github.com/repos/${REPOSITORY_NAME}/releases/tags/${TAG}" --header "authorization: Bearer ${GITHUB_TOKEN}" | jq -r '.id')
```
Comment (Konard) [inline MultiProjectRepository/publish-release.sh:13]:
```suggestion
TAG_ID=$(curl --request GET --url "https://api.github.com/repos/${GITHUB_REPOSITORY}/releases/tags/${TAG}" --header "authorization: Bearer ${GITHUB_TOKEN}" | jq -r '.id')
```

File: MultiProjectRepository/publish-release.sh [hunk 5]
```diff
@@ -1,59 +1,34 @@
...
-        NAME_BUFFER="$NAME_BUFFER, [C++] $CPP_PACKAGE_VERSION"
-        PACKAGE_RELEASE_NOTES_STRING_BUFFER="$PACKAGE_RELEASE_NOTES_STRING_BUFFER$SEPARATOR$SEPARATOR$CPP_PACKAGE_URL$SEPARATOR$CPP_PACKAGE_RELEASE_NOTES"
-    fi
-fi
-
-TAG_ID=$(curl --request GET --url "https://api.github.com/repos/${GITHUB_REPOSITORY}/releases/tags/${TAG_BUFFER}" --header "authorization: Bearer ${GITHUB_TOKEN}" | jq -r '.id')
-
-if [ "$TAG_ID" != "null" ]; then
-    echo "Release with the same tag already published."
+if [ "$tag_id" != "null" ]; then
```
Comment (Konard) [inline MultiProjectRepository/publish-release.sh:15]:
```suggestion
if [ "$TAG_ID" != "null" ]; then
```

**Review by Konard — CHANGES_REQUESTED** [review]

File: MultiProjectRepository/cpp-publish-release.sh [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+if [ ! -f "CPP_PACKAGE_VERSION.txt" ]
+then
+    echo "CPP_PACKAGE_VERSION.txt not found."
+    exit 1
+fi
+
+PACKAGE_VERSION=$(<CPP_PACKAGE_VERSION.txt)
+PACKAGE_RELEASE_NOTES=$(<CPP_PACKAGE_RELEASE_NOTES.txt)
+PACKAGE_URL="https://www.nuget.org/packages/Platform.$GITHUB_REPOSITORY.TemplateLibrary/$PACKAGE_VERSION"
```
Comment (Konard) [inline MultiProjectRepository/cpp-publish-release.sh:16]:
```suggestion
PACKAGE_URL="https://www.nuget.org/packages/Platform.$REPOSITORY_NAME.TemplateLibrary/$PACKAGE_VERSION"
```

File: MultiProjectRepository/csharp-publish-release.sh [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+
+if [ ! -f "CSHARP_PACKAGE_VERSION.txt" ]
+then
+    echo "CSHARP_PACKAGE_VERSION.txt not found."
+    exit 1
+fi
+
+PACKAGE_VERSION=$(<CSHARP_PACKAGE_VERSION.txt)
+PACKAGE_RELEASE_NOTES=$(<CSHARP_PACKAGE_RELEASE_NOTES.txt)
+PACKAGE_URL="https://www.nuget.org/packages/Platform.$GITHUB_REPOSITORY/$CSHARP_PACKAGE_VERSION"
```
Comment (Konard) [inline MultiProjectRepository/csharp-publish-release.sh:16]:
```suggestion
PACKAGE_URL="https://www.nuget.org/packages/Platform.$REPOSITORY_NAME/$CSHARP_PACKAGE_VERSION"
```

**Review by Konard — CHANGES_REQUESTED** [review]

File: MultiProjectRepository/rust-publish-release.sh [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+
+if [ ! -f "RUST_PACKAGE_VERSION.txt" ]
+then
+    echo "RUST_PACKAGE_VERSION.txt not found."
+    exit 1
+fi
+
+PACKAGE_VERSION=$(<RUST_PACKAGE_VERSION.txt)
+PACKAGE_RELEASE_NOTES=$(<RUST_PACKAGE_RELEASE_NOTES.txt)
+PACKAGE_URL="https://crates.io/crates/doublets/$PACKAGE_VERSION"
```
Comment (Konard) [inline MultiProjectRepository/rust-publish-release.sh:16]: You should read package name from the file too.

File: MultiProjectRepository/rust-publish-release.sh [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+"
+
+if [ ! -f "RUST_PACKAGE_VERSION.txt" ]
+then
+    echo "RUST_PACKAGE_VERSION.txt not found."
+    exit 1
+fi
+
+PACKAGE_VERSION=$(<RUST_PACKAGE_VERSION.txt)
+PACKAGE_RELEASE_NOTES=$(<RUST_PACKAGE_RELEASE_NOTES.txt)
```
Comment (Konard) [inline MultiProjectRepository/rust-publish-release.sh:15]:
```suggestion
```

There is no release notes for Rust.

File: MultiProjectRepository/rust-publish-release.sh [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+PACKAGE_URL="https://crates.io/crates/doublets/$PACKAGE_VERSION"
+
+TAG="rust_$PACKAGE_VERSION"
+RELEASE_NAME="[Rust] $PACKAGE_VERSION"
+RELEASE_BODY="$PACKAGE_URL$SEPARATOR$PACKAGE_RELEASE_NOTES"
```
Comment (Konard) [inline MultiProjectRepository/rust-publish-release.sh:20]:
```suggestion
RELEASE_BODY="$PACKAGE_URL"
```

File: MultiProjectRepository/rust-publish-release.sh (hunk 8 above)
Comment (FreePhoenix888) [reply MultiProjectRepository/rust-publish-release.sh:16]: > You should read package name from the file too.

How?

Comment (Konard) [reply MultiProjectRepository/rust-publish-release.sh:15]: https://github.com/linksplatform/Scripts/blob/main/MultiProjectRepository/read_rust_package_info.sh#L9 the name of the package is in this file.

**Review by Konard — APPROVED** [review]

**Pull request merged**